From cf0494cfd348764e2a7d17781436bd95a6fae1ef Mon Sep 17 00:00:00 2001
From: Darren Etheridge <detheridge@ti.com>
Date: Sun, 16 Jun 2024 21:45:23 -0500
Subject: [PATCH 178/222] media: platform: img: Add the vxe-vxd driver into the
 build infra

Update Kconfig to account for the IMG Encoder/Decoder drivers and enable
build in relevant Makefiles.

Signed-off-by: Darren Etheridge <detheridge@ti.com>
Signed-off-by: Brandon Brnich <b-brnich@ti.com>
---

diff --git a/drivers/media/platform/imagination/vxe-vxd/Kconfig b/drivers/media/platform/imagination/Kconfig
new file mode 100644
index 000000000000..562753d7febf
--- /dev/null
+++ b/drivers/media/platform/imagination/Kconfig
@@ -0,0 +1,36 @@
+# SPDX-License-Identifier: GPL-2.0-only
+
+comment "IMG media platform drivers"
+
+config VIDEO_IMG_VXD_DEC
+  tristate "IMG VXD DEC (Video Decoder) driver"
+	depends on V4L_MEM2MEM_DRIVERS
+  depends on VIDEO_DEV
+  depends on ARCH_K3 || COMPILE_TEST
+  depends on V4L_PLATFORM_DRIVERS
+  select VIDEOBUF2_CORE
+  select VIDEOBUF2_DMA_CONTIG
+  select VIDEOBUF2_DMA_SG
+  select V4L2_MEM2MEM_DEV
+  help
+    This is an IMG VXD DEC V4L2 driver that adds support for the
+    Imagination D5520 (Video Decoder) hardware as used on the Texas
+    Instruments Jacinto and Sitara devices.
+    The module name when built is vxd-dec.
+
+config VIDEO_IMG_VXE_ENC
+  tristate "IMG VXE ENC (Video Encoder) driver"
+	depends on V4L_MEM2MEM_DRIVERS
+  depends on VIDEO_DEV
+  depends on ARCH_K3 || COMPILE_TEST
+  depends on V4L_PLATFORM_DRIVERS
+  select VIDEOBUF2_CORE
+  select VIDEOBUF2_DMA_CONTIG
+  select VIDEOBUF2_DMA_SG
+  select V4L2_MEM2MEM_DEV
+  help
+    This is an IMG VXE ENC V4L2 driver that adds support for the
+    Imagination VXE384 (Video Encoder) hardware as used on the Texas
+    Instruments Jacinto and Sitara devices.
+    The module name when built is vxe-enc.
+
diff --git a/drivers/media/platform/imagination/vxe-vxd/Makefile b/drivers/media/platform/imagination/Makefile
new file mode 100644
index 000000000000..57a8d7ba664d
--- /dev/null
+++ b/drivers/media/platform/imagination/Makefile
@@ -0,0 +1,2 @@
+# SPDX-License-Identifier: GPL-2.0-only
+obj-y += vxe-vxd/
